http/oidc/src/test/java/org/wildfly/security/http/oidc/OidcTest.java

```diff
@@ -367,6 +367,21 @@ public void testRequestObjectConfigMissingENCValue() throws Exception {
         testRequestObjectInvalidConfiguration(getOidcConfigurationInputStreamWithoutEncValue(REQUEST.getValue(), RSA_OAEP), RequestObjectErrorType.MISSING_ENC_VALUE);
     }
 
+    @Test
+    // Generate an invalid sessionRandomValue so that the nonce check fails
+    public void testSessionIdNonceMismatch() throws Exception {
```

Comment:
Would be good to rename to `testRequestParameterNonceMismatch` and to also add one more test method called `testRequestUriParameterNonceMismatch` that uses `getOidcConfigurationInputStreamWithRequestParameter(REQUEST_URI.getValue()...`

Reply:
done